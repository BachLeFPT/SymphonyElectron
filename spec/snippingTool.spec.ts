import { shallow } from 'enzyme';
import * as React from 'react';
import SnippingTool from '../src/renderer/components/snipping-tool';
import { ipcRenderer } from './__mocks__/electron';

describe('Snipping Tool', () => {
  const snippingToolLabel = 'snipping-tool-data';
  const onLabelEvent = 'on';

  it('should render correctly', () => {
    const wrapper = shallow(React.createElement(SnippingTool));
    expect(wrapper).toMatchSnapshot();
  });

  it('should call `snipping-tool-data` event when component is mounted', () => {
    const spy = jest.spyOn(ipcRenderer, onLabelEvent);
    shallow(React.createElement(SnippingTool));
    expect(spy).toBeCalledWith(snippingToolLabel, expect.any(Function));
  });

  it('should render pen color picker when clicked on pen', () => {
    const wrapper = shallow(React.createElement(SnippingTool));
    wrapper.find('[data-testid="pen-button"]').simulate('click');
    expect(wrapper.find('[data-testid="pen-colorpicker"]').exists()).toBe(true);
  });

  it('should render highlight color picker when clicked on highlight', () => {
    const wrapper = shallow(React.createElement(SnippingTool));
    wrapper.find('[data-testid="highlight-button"]').simulate('click');
    expect(wrapper.find('[data-testid="highlight-colorpicker"]').exists()).toBe(true);
  });
});
