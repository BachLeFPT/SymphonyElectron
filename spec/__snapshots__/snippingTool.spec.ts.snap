// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snipping Tool should render correctly 1`] = `
<div
  className="SnippingTool"
  lang="en-US"
>
  <header>
    <div
      className="DrawActions"
    >
      <button
        className="ActionButton"
        onClick={[Function]}
        style={
          Object {
            "border": "2px solid rgba(0, 142, 255, 1)",
          }
        }
      >
        <img
          src="../renderer/assets/snip-draw.svg"
        />
      </button>
      <button
        className="ActionButton"
        onClick={[Function]}
      >
        <img
          src="../renderer/assets/snip-highlight.svg"
        />
      </button>
      <button
        className="ActionButton"
        onClick={[Function]}
      >
        <img
          src="../renderer/assets/snip-erase.svg"
        />
      </button>
    </div>
    <div
      className="ClearActions"
    >
      <button
        className="ClearButton"
        onClick={[Function]}
      >
        Clear
      </button>
    </div>
  </header>
  ;
  <main>
    <div>
      <img
        alt="Screen snippet"
        className="SnippetImage"
        height={600}
        src="Screen-Snippet"
        width={800}
      />
    </div>
  </main>
  <footer>
    <button
      onClick={[Function]}
    >
      Done
    </button>
  </footer>
</div>
`;
